File: Assets/AltTester/Runtime/AltDriver/Commands/InputActions/AltTapCoordinates.cs
/*
    Copyright(C) 2025 Altom Consulting

    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.

    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with this program. If not, see <https://www.gnu.org/licenses/>.
*/

namespace AltTester.AltTesterSDK.Driver.Commands
{
    public class AltTapCoordinates : AltBaseCommand
    {
        AltTapCoordinatesParams cmdParams;
        public AltTapCoordinates(IDriverCommunication commHandler, AltVector2 coordinates, int count, float interval, bool wait) : base(commHandler)
        {
            cmdParams = new AltTapCoordinatesParams(coordinates, count, interval, wait);
        }
        public void Execute()
        {
            CommHandler.Send(cmdParams);
            string data = CommHandler.Recvall<string>(cmdParams);
            ValidateResponse("Ok", data);
            if (cmdParams.wait)
            {
                data = CommHandler.Recvall<string>(cmdParams); ;
                ValidateResponse("Finished", data);
            }
        }
    }
}

File: Bindings~/java/src/main/java/com/alttester/Commands/InputActions/AltTapCoordinates.java
/*
    Copyright(C) 2026 Altom Consulting
*/

package com.alttester.Commands.InputActions;

import com.alttester.Commands.AltBaseCommand;
import com.alttester.IMessageHandler;

public class AltTapCoordinates extends AltBaseCommand {
  private AltTapClickCoordinatesParams parameters;

  public AltTapCoordinates(
      IMessageHandler messageHandler, AltTapClickCoordinatesParams parameters) {
    super(messageHandler);
    this.parameters = parameters;
    this.parameters.setCommandName("tapCoordinates");
  }

  public void Execute() {
    SendCommand(parameters);
    String data = recvall(parameters, String.class);
    validateResponse("Ok", data);
    if (parameters.getWait()) {
      data = recvall(parameters, String.class);
      validateResponse("Finished", data);
    }
  }
}
